### PR TITLE
Release 0.9.4

### LINKED ISSUE #586: Solver diagnostics for adjoint sensitivities not written to ReturnData for the last timepoint
... since `BackwardProblem::handleDatapointB`, and thus, `Solver::getDiagnosisB` is never called.

Should they also be reported if the next timepoint is never reached? (CVODES mxsteps error)

### PR DESCRIPTION
**Pull request by dweindl**
Minor fixes only:
- fix(core) Get solver diagnostics for first(last) timepoint (#588) (Closes #586)
- fix(ci) Fix autodeploy (Closes #589)

### CODE REVIEW

File: include/amici/solver.h [hunk 1]
```diff
@@ -75,7 +75,7 @@ class Solver {
      * @param model pointer to the model object
      */
 
-    void setupAMIB(BackwardProblem *bwd, Model *model);
+    void setupB(BackwardProblem *bwd, Model *model);
```
Comment (yannikschaelte) [inline include/amici/solver.h]: why shorter? + maybe remove the empty line before?

File: src/backwardproblem.cpp [hunk 2]
```diff
@@ -178,36 +127,19 @@ void BackwardProblem::handleEventB(const int iroot) {
     model->updateHeavisideB(&rootidx[iroot * model->ne]);
 }
 
-/**
- * handleDataPoint executes everything necessary for the handling of data
- * points for the backward problems
- *
- * @param it index of data point @type int
- */
+
```
Comment (yannikschaelte) [inline src/backwardproblem.cpp]: I assume these docs all went to the h files?

File: include/amici/solver.h (hunk 1 above)
Comment (dweindl) [reply include/amici/solver.h]: AMI prefix is a relict from C times. We have amici namespace now.

File: src/backwardproblem.cpp (hunk 2 above)
Comment (dweindl) [reply src/backwardproblem.cpp]: Assumption correct.

**Pull request merged**